check-openmp: some test cases failed on arm64/MacOS

Run test cases of OpenMP on M1 Mac with clang 14.0.3, getting the following failed test cases

>   libomp :: barrier/omp_barrier.c
>   libomp :: ompt/synchronization/masked.c
>   libomp :: ompt/synchronization/master.c
>   libomp :: parallel/omp_parallel_if.c
> 
> 
> Testing Time: 209.32s
>   Unsupported      :  23
>   Passed           : 285
>   Expectedly Failed:   1
>   Failed           :   4


Steps to reproduce:

```bash
git clone -n https://github.com/llvm/llvm-project.git
mkdir llvm-project/build
cd llvm-project/build
cmake -G Ninja ../llvm \
   -DLLVM_ENABLE_PROJECTS="openmp;clang" \
   -DLLVM_TARGETS_TO_BUILD="host" \
   -DCMAKE_BUILD_TYPE=Release \
   -DLLVM_ENABLE_ASSERTIONS=ON \
   -DLLVM_ENABLE_RTTI=ON \
   -DLLVM_ENABLE_LIBEDIT=OFF \
   -DLLVM_BUILD_EXAMPLES=ON

cmake --build . --target check-openmp
```


Detailed console output:

```
(base) macbook@host-130-148 build % cmake --build . --target check-openmp
[2687/2688] Running OpenMP tests
FAIL: libomp :: barrier/omp_barrier.c (104 of 313)
******************** TEST 'libomp :: barrier/omp_barrier.c' FAILED ********************
Script:
--
: 'RUN: at line 1';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp -lm && /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp
: 'RUN: at line 2';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp -lm && env KMP_BLOCKTIME=infinite /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp
: 'RUN: at line 3';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp -lm && env KMP_PLAIN_BARRIER_PATTERN='hierarchical,hierarchical' KMP_FORKJOIN_BARRIER_PATTERN='hierarchical,hierarchical' /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp
: 'RUN: at line 4';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp -lm && env KMP_BLOCKTIME=infinite KMP_PLAIN_BARRIER_PATTERN='hierarchical,hierarchical' KMP_FORKJOIN_BARRIER_PATTERN='hierarchical,hierarchical' /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp
: 'RUN: at line 5';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp -lm && env KMP_PLAIN_BARRIER_PATTERN='dist,dist' KMP_FORKJOIN_BARRIER_PATTERN='dist,dist' KMP_REDUCTION_BARRIER_PATTERN='dist,dist' /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp
: 'RUN: at line 6';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp -lm && env KMP_BLOCKTIME=infinite KMP_PLAIN_BARRIER_PATTERN='dist,dist' KMP_FORKJOIN_BARRIER_PATTERN='dist,dist' KMP_REDUCTION_BARRIER_PATTERN='dist,dist' /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp
--
Exit Code: -6

Command Output (stdout):
--
$ ":" "RUN: at line 1"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp" "-lm"
$ "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp"
$ ":" "RUN: at line 2"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp" "-lm"
$ "env" "KMP_BLOCKTIME=infinite" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp"
$ ":" "RUN: at line 3"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp" "-lm"
$ "env" "KMP_PLAIN_BARRIER_PATTERN=hierarchical,hierarchical" "KMP_FORKJOIN_BARRIER_PATTERN=hierarchical,hierarchical" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp"
$ ":" "RUN: at line 4"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp" "-lm"
$ "env" "KMP_BLOCKTIME=infinite" "KMP_PLAIN_BARRIER_PATTERN=hierarchical,hierarchical" "KMP_FORKJOIN_BARRIER_PATTERN=hierarchical,hierarchical" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp"
$ ":" "RUN: at line 5"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/barrier/omp_barrier.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp" "-lm"
$ "env" "KMP_PLAIN_BARRIER_PATTERN=dist,dist" "KMP_FORKJOIN_BARRIER_PATTERN=dist,dist" "KMP_REDUCTION_BARRIER_PATTERN=dist,dist" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/barrier/Output/omp_barrier.c.tmp"
# command stderr:
OMP: Error #111: Memory allocation failed.

error: command failed with exit status: -6

--

********************
FAIL: libomp :: ompt/synchronization/masked.c (122 of 313)
******************** TEST 'libomp :: ompt/synchronization/masked.c' FAILED ********************
Script:
--
: 'RUN: at line 1';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/masked.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/masked.c.tmp -lm && /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/masked.c.tmp | tee /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/masked.c.tmp.out | /Users/macbook/GitHub/llvm-project/build/./bin/FileCheck /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/masked.c
--
Exit Code: 1

Command Output (stdout):
--
$ ":" "RUN: at line 1"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/masked.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/masked.c.tmp" "-lm"
$ "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/masked.c.tmp"
$ "tee" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/masked.c.tmp.out"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/FileCheck" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/masked.c"
# command stderr:
/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/masked.c:38:11: error: CHECK: expected string not found in input
// CHECK: {{^}}[[MASTER_ID]]: current_address={{.*}}[[RETURN_ADDRESS_END]]
          ^
<stdin>:8:117: note: scanning from here
281474976710658: ompt_event_masked_end: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x1048811cc
                                                                                                                    ^
<stdin>:8:117: note: with "MASTER_ID" equal to "281474976710658"
281474976710658: ompt_event_masked_end: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x1048811cc
                                                                                                                    ^
<stdin>:8:117: note: with "RETURN_ADDRESS_END" equal to "0x1048811cc"
281474976710658: ompt_event_masked_end: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x1048811cc
                                                                                                                    ^
<stdin>:9:13: note: possible intended match here
281474976710658: current_address=0x1048811d4 or 0x1048811d0
            ^

Input file: <stdin>
Check file: /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/masked.c

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: 0: NULL_POINTER=0x0 
            2: 281474976710658: ompt_event_thread_begin: thread_type=ompt_thread_initial=1, thread_id=281474976710658 
            3: 281474976710658: ompt_event_initial_task_begin: parallel_id=281474976710660, task_id=281474976710659, actual_parallelism=1, index=1, flags=1 
            4: 281474976710658: ompt_event_parallel_begin: parent_task_id=281474976710659, parent_task_frame.exit=0x0, parent_task_frame.reenter=0x16b583640, parallel_id=281474976710661, requested_team_size=2, codeptr_ra=0x1048810a0, invoker=2 
            5: 281474976710658: ompt_event_implicit_task_begin: parallel_id=281474976710661, task_id=281474976710662, team_size=2, thread_num=0 
            6: 281474976710658: ompt_event_masked_begin: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x104881134 
            7: 281474976710658: fuzzy_address=0x1048810 or 0x1048811 or 0x1048812 or 0x1048813 (0x10488114c) 
            8: 281474976710658: ompt_event_masked_end: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x1048811cc 
check:38'0                                                                                                                         X error: no match found
check:38'1                                                                                                                           with "MASTER_ID" equal to "281474976710658"
check:38'2                                                                                                                           with "RETURN_ADDRESS_END" equal to "0x1048811cc"
            9: 281474976710658: current_address=0x1048811d4 or 0x1048811d0 
check:38'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:38'3                 ?                                                possible intended match
           10: 281474976710658: ompt_event_barrier_begin: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x1048810a0 
check:38'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           11: 281474976710658: task level 0: parallel_id=281474976710661, task_id=281474976710662, exit_frame=0x0, reenter_frame=0x0, task_type=ompt_task_implicit|ompt_task_undeferred=134217730, thread_num=0 
check:38'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           12: 281474976710658: ompt_event_wait_barrier_begin: parallel_id=281474976710661, task_id=281474976710662, codeptr_ra=0x1048810a0 
check:38'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           13: 562949953421313: ompt_event_thread_begin: thread_type=ompt_thread_worker=2, thread_id=562949953421313 
check:38'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           14: 562949953421313: ompt_event_implicit_task_begin: parallel_id=281474976710661, task_id=562949953421314, team_size=2, thread_num=1 
check:38'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            .
            .
            .
>>>>>>

error: command failed with exit status: 1

--

********************
FAIL: libomp :: ompt/synchronization/master.c (125 of 313)
******************** TEST 'libomp :: ompt/synchronization/master.c' FAILED ********************
Script:
--
: 'RUN: at line 1';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/master.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/master.c.tmp -lm && /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/master.c.tmp | tee /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/master.c.tmp.out | /Users/macbook/GitHub/llvm-project/build/./bin/FileCheck /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/master.c
--
Exit Code: 1

Command Output (stdout):
--
$ ":" "RUN: at line 1"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/master.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/master.c.tmp" "-lm"
$ "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/master.c.tmp"
$ "tee" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/ompt/synchronization/Output/master.c.tmp.out"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/FileCheck" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/master.c"
# command stderr:
/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/master.c:89:11: error: CHECK: expected string not found in input
// CHECK: {{^}}[[MASTER_ID]]: current_address={{.*}}[[RETURN_ADDRESS_END]]
          ^
<stdin>:4:49: note: scanning from here
0: ompt_event_master_end: codeptr_ra=0x10268fa08
                                                ^
<stdin>:4:49: note: with "MASTER_ID" equal to "0"
0: ompt_event_master_end: codeptr_ra=0x10268fa08
                                                ^
<stdin>:4:49: note: with "RETURN_ADDRESS_END" equal to "0x10268fa08"
0: ompt_event_master_end: codeptr_ra=0x10268fa08
                                                ^
<stdin>:5:1: note: possible intended match here
0: current_address=0x10268fa10 or 0x10268fa0c
^

Input file: <stdin>
Check file: /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt/synchronization/master.c

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: 0: NULL_POINTER=0x0 
            2: 0: ompt_event_master_begin: codeptr_ra=0x10268f970 
            3: 0: fuzzy_address=0x10268f8 or 0x10268f9 or 0x10268fa or 0x10268fb (0x10268f988) 
            4: 0: ompt_event_master_end: codeptr_ra=0x10268fa08 
check:89'0                                                     X error: no match found
check:89'1                                                       with "MASTER_ID" equal to "0"
check:89'2                                                       with "RETURN_ADDRESS_END" equal to "0x10268fa08"
            5: 0: current_address=0x10268fa10 or 0x10268fa0c 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:89'3     ?                                              possible intended match
            6: 0: current_address=0x10268fa10 or 0x10268fa0c 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            7: 0: x=1 
check:89'0     ~~~~~~~
>>>>>>

error: command failed with exit status: 1

--

********************
FAIL: libomp :: parallel/omp_parallel_if.c (198 of 313)
******************** TEST 'libomp :: parallel/omp_parallel_if.c' FAILED ********************
Script:
--
: 'RUN: at line 1';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt /Users/macbook/GitHub/llvm-project/openmp/runtime/test/parallel/omp_parallel_if.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp -lm && /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp
: 'RUN: at line 2';   /Users/macbook/GitHub/llvm-project/build/./bin/clang -fopenmp   -I /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test -L /Users/macbook/GitHub/llvm-project/build/./lib  -fno-omit-frame-pointer -Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I /Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt -fopenmp-enable-irbuilder /Users/macbook/GitHub/llvm-project/openmp/runtime/test/parallel/omp_parallel_if.c -o /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp -lm && /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp
--
Exit Code: -11

Command Output (stdout):
--
$ ":" "RUN: at line 1"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/parallel/omp_parallel_if.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp" "-lm"
$ "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp"
$ ":" "RUN: at line 2"
$ "/Users/macbook/GitHub/llvm-project/build/./bin/clang" "-fopenmp" "-I" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/src" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test" "-L" "/Users/macbook/GitHub/llvm-project/build/./lib" "-fno-omit-frame-pointer" "-Wl,-rpath,/Users/macbook/GitHub/llvm-project/build/./lib" "-isysroot" "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" "-I" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/ompt" "-fopenmp-enable-irbuilder" "/Users/macbook/GitHub/llvm-project/openmp/runtime/test/parallel/omp_parallel_if.c" "-o" "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp" "-lm"
$ "/Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test/parallel/Output/omp_parallel_if.c.tmp"
note: command had no output on stdout or stderr
error: command failed with exit status: -11

--

********************
********************
Failed Tests (4):
  libomp :: barrier/omp_barrier.c
  libomp :: ompt/synchronization/masked.c
  libomp :: ompt/synchronization/master.c
  libomp :: parallel/omp_parallel_if.c


Testing Time: 209.32s
  Unsupported      :  23
  Passed           : 285
  Expectedly Failed:   1
  Failed           :   4
FAILED: projects/openmp/CMakeFiles/check-openmp /Users/macbook/GitHub/llvm-project/build/projects/openmp/CMakeFiles/check-openmp 
cd /Users/macbook/GitHub/llvm-project/build/projects/openmp && /Users/macbook/miniconda3/bin/python3.10 /Users/macbook/GitHub/llvm-project/build/./bin/llvm-lit -sv /Users/macbook/GitHub/llvm-project/build/projects/openmp/runtime/test
ninja: build stopped: subcommand failed.
```

--- a/openmp/runtime/src/kmp_csupport.cpp
+++ b/openmp/runtime/src/kmp_csupport.cpp
@@ -343,7 +343,6 @@ Perform a fork only if the condition is true.
 void __kmpc_fork_call_if(ident_t *loc, kmp_int32 argc, kmpc_micro microtask,
                          kmp_int32 cond, void *args) {
   int gtid = __kmp_entry_gtid();
-  int zero = 0;
   if (cond) {
     if (args)
       __kmpc_fork_call(loc, argc, microtask, args);
@@ -352,10 +351,29 @@ void __kmpc_fork_call_if(ident_t *loc, kmp_int32 argc, kmpc_micro microtask,
   } else {
     __kmpc_serialized_parallel(loc, gtid);
 
+#ifdef OMPT_SUPPORT
+    void *exit_frame_ptr;
+#endif
+
     if (args)
-      microtask(&gtid, &zero, args);
+      __kmp_invoke_microtask(VOLATILE_CAST(microtask_t) microtask, gtid,
+                             /*npr=*/0,
+                             /*argc=*/1, &args
+#ifdef OMPT_SUPPORT
+                             ,
+                             &exit_frame_ptr
+#endif
+      );
     else
-      microtask(&gtid, &zero);
+      __kmp_invoke_microtask(VOLATILE_CAST(microtask_t) microtask, gtid,
+                             /*npr=*/0,
+                             /*argc=*/0,
+                             /*args=*/nullptr
+#ifdef OMPT_SUPPORT
+                             ,
+                             &exit_frame_ptr
+#endif
+      );
 
     __kmpc_end_serialized_parallel(loc, gtid);
   }

